docs/csharp/how-to/compare-strings.md

```diff
@@ -0,0 +1,86 @@
+
+[!code-csharp-interactive[Demonstrating string interning](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#4)]
+
+## Ordinal sorting and searching strings
+
+The following examples show how to sort and search for strings in an array in a culture-sensitive manner by using the static <xref:System.Array> methods that take a <xref:System.StringComparer?displayProperty=nameWithType> parameter.
+
+This example shows how to search an array of strings using an ordinal search: 
+
+[!code-csharp-interactive[Sorting an array of strings](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#5)]
+
+Once the array is sorted, you can search for entries using a binary search. A binary search starts in the middle of the collection to determine which half of the collection would contain the sought string. Each subsequent comparison subdivides the remaining part of the collection in half. The following sample demonstrates the importance of using the same comparison rules for both sorting and searching. The array is sorted using <xref:System.StringComparer.Ordinal?displayProperty=nameWithType>. However, the search uses <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType>. The local function `ShowWhere` displays information about where the string was found. If the string was not found, the returned value indicates where it would be if it were found.
+
+[!code-csharp-interactive[Searching in a sorted array](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#6)]
+
+## Case sensitive and culture sensitive sorting and searching
+
+The following code uses the <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> collection class to store strings. The strings are sorted using the <xref:System.Collections.Generic.List%601.Sort%2A?displayProperty=nameWithType> method. This method needs a delegate that compares and orders two strings. The <xref:System.String.CompareTo%2A?displayProperty=nameWithType> method provides that comparison function. Run the sample and observe the order. This sort operation is both case sensitive and culture sensitive. You would use the static <xref:System.String.Compare%2A?displayProperty=nameWithType> methods to specify different comparison rules.
+
+[!code-csharp-interactive[Sorting a list of strings](../../../samples/snippets/csharp/how-to/strings/CompareStrings.cs#7)]
+
+Once sorted, the list of strings can be searched using a binary search. The following sample shows how to search the sorted listed using the same comparison function. The local function `ShowWhere` shows where the sought text is or would be:
```

Comment:
Binary search is not appropriate here. If you're doing a binary search on culture-sensitive strings, searches are likely to fail for strings that contain characters outside the ASCII range. The search method should correspond to the sorting method.

Reply:
@rpetrusha This is the one comment that confuses me. What search method should be used when the collection is sorted in a culture-sensitive manner? Is it necessary to do a linear search?